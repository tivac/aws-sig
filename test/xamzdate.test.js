"use strict";

const { headerXAmzDate : date } = require("../src/aws-sig.js");

describe("aws-sig core API", () => {
    afterEach(() => jest.restoreAllMocks());

    it("should export X-Amz-Date creation", () => {
        expect(typeof date).toBe("function");
    });

    it("should convert JS Date objects into X-Amz-Date format", () => {
        const input = new Date("Tue, 15 Nov 1994 08:12:31 GMT");

        expect(date(input)).toMatchSnapshot();
    });

    it("should use Date.now() if a date isn't specified", () => {
        const mock = jest.spyOn(Date, "now")
            .mockImplementation(() => Date.parse("Tue, 15 Nov 1994 08:12:31 GMT"));
        
        expect(date()).toMatchSnapshot();

        expect(mock).toHaveBeenCalled();
        expect(mock).toHaveReturned();
    });
});
