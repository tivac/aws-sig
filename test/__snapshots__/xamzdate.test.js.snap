// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aws-sig core API should convert JS Date objects into X-Amz-Date format 1`] = `"19941115T081231Z"`;

exports[`aws-sig core API should use Date.now() if a date isn't specified 1`] = `"19941115T081231Z"`;
